protocol/v2/types/ssvshare.go

```diff
@@ -58,16 +60,22 @@ func (s *SSVShare) SetFeeRecipient(feeRecipient bellatrix.ExecutionAddress) {
 	s.FeeRecipientAddress = feeRecipient
 }
 
+// CommitteeID safely retrieves or computes the CommitteeID.
 func (s *SSVShare) CommitteeID() spectypes.CommitteeID {
-	if s.committeeID != nil {
-		return *s.committeeID
+	// Load the current value of committeeID atomically.
+	if ptr := s.committeeID.Load(); ptr != nil {
+		return *ptr
 	}
+
+	// Compute the CommitteeID since it's not yet set.
 	ids := make([]spectypes.OperatorID, len(s.Share.Committee))
 	for i, v := range s.Share.Committee {
 		ids[i] = v.Signer
 	}
 	id := ComputeCommitteeID(ids)
-	s.committeeID = &id
+
+	// Create a new pointer and store it atomically.
+	s.committeeID.Store(&id)
```

Comment:
committeeID should never change, isn't it better to just have a `New` function to `SSVShare` which sets the committeeID?

Reply:
@y0sher unfortunately Go has weak constructors, so that would pass the responsibility of guaranteeing `Share` is never created without `New` to us, for not much benefit (as far as I can tell)

after we decouple from spec, we can refactor `Share` to an interface backed by a struct with unexported fields to enforce that